kyo-core/shared/src/main/scala/kyo/Clock.scala

```diff
@@ -128,31 +284,39 @@ object Clock:
       *   A new Deadline instance
       */
     def deadline(duration: Duration)(using Frame): Deadline < IO =
-        local.use(_.deadline(duration))
-
-    /** Creates a new Clock instance from an Unsafe implementation.
-      *
-      * @param u
-      *   The Unsafe implementation
-      * @return
-      *   A new Clock instance
-      */
-    def apply(u: Unsafe): Clock =
-        new Clock:
-            def now(using Frame): Instant < IO =
-                IO.Unsafe(u.now())
-            def unsafe: Unsafe = u
+        use(_.deadline(duration))
 
     /** WARNING: Low-level API meant for integrations, libraries, and performance-sensitive code. See AllowUnsafe for more details. */
     abstract class Unsafe:
+        import AllowUnsafe.embrace.danger
+
         def now()(using AllowUnsafe): Instant
 
-        def stopwatch()(using AllowUnsafe): Stopwatch.Unsafe = Stopwatch.Unsafe(now(), this)
+        private[kyo] def sleep(duration: Duration): Fiber.Unsafe[Nothing, Unit]
+
+        final def stopwatch()(using AllowUnsafe): Stopwatch.Unsafe = Stopwatch.Unsafe(now(), this)
 
-        def deadline(duration: Duration)(using AllowUnsafe): Deadline.Unsafe =
+        final def deadline(duration: Duration)(using AllowUnsafe): Deadline.Unsafe =
             if !duration.isFinite then Deadline.Unsafe(Maybe.empty, this)
             else Deadline.Unsafe(Maybe(now() + duration), this)
 
-        def safe: Clock = Clock(this)
+        final def safe: Clock = Clock(this)
+    end Unsafe
+
+    object Unsafe:
+        def apply(executor: ScheduledExecutorService)(using AllowUnsafe): Unsafe =
+            new Unsafe:
+                def now()(using AllowUnsafe) = Instant.fromJava(java.time.Instant.now())
+                def sleep(duration: Duration) =
+                    Promise.Unsafe.fromIOPromise {
+                        new IOPromise[Nothing, Unit] with Callable[Unit]:
```

Comment:
Any reason to use `Callable[Unit]` vs `Runnable`?

Reply:
I had changed it to `Runnable` but then I remembered why I used `Callable`. It's what's supported by the ScalaJS stubs:

```
[info] compiling 27 Scala sources to /home/runner/work/kyo/kyo/kyo-prelude/js/target/scala-3.5.2/test-classes ...
[error] -- [E007] Type Mismatch Error: /home/runner/work/kyo/kyo/kyo-core/shared/src/main/scala/kyo/Clock.scala:516:57 
[error] 516 |                            val task = executor.schedule(this, duration.toNanos, TimeUnit.NANOSECONDS)
[error]     |                                                         ^^^^
[error]     |Found:    ($anon.this : kyo.scheduler.IOPromise[Nothing, Unit] with Runnable {...})
[error]     |Required: java.util.concurrent.Callable[A]
[error]     |
[error]     |where:    A is a type variable
[error]     |
[error]     | longer explanation available when compiling with `-explain`
[error] one error found
```